### PR TITLE
feat: less to component token for Timeline

### PR DESCRIPTION
**Pull request by jrr997**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

ref https://github.com/ant-design/ant-design/issues/41884
### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add Timeline Component Token. |
| 🇨🇳 Chinese | 添加 Timeline 的 Component Token。|

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 3709f54</samp>

This pull request enhances the Timeline component by adding a custom component token interface that allows the user to customize some style properties. It also updates the documentation and adds a demo to show how to use the custom tokens. It modifies the files `components/timeline/style/index.ts`, `docs/react/migrate-less-variables.en-US.md`, `docs/react/migrate-less-variables.zh-CN.md`, `components/timeline/demo/componentToken-debug.md`, `components/timeline/demo/componentToken-debug.tsx`, `components/timeline/index.en-US.md`, and `components/timeline/index.zh-CN.md`.

### <samp>🤖 Generated by Copilot at 3709f54</samp>

*  Add the custom component token demo of the Timeline component ([link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-4a629d0b8e7843851b9dc711f55be9e123869571f9f4449e5cc8ad488be8d1beR1-R7), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-2334ae8d5878b3323dbbd00cd88235544bbc5599ebb5d9529c1a596933064b50R1-R36), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-942630313a2a8e88a851826936211799361fc1be091a081dfe21c0476d03a6feR44), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-674d1128c0370312222089b3797367cb6a152a35bf442d37e266300641d1f1d6R45))
*  Add the custom component token interface to the Timeline component style file and rename or remove the deprecated token properties ([link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L6-R17), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L30-R33), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L36-R42), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L55-R61), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L84-R92), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L100-R103), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L130-R133), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L158-R166), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L173-R176), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L187-R190), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L199-R202), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-033ff0b0fb35d11b0031fce6bb9dba6e8d00910e9dbf81a620c42f904a18d602L226-R247))
*  Update the migrate less variables guide for the Timeline component and remove the empty lines from the documentation ([link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL42), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL56), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL137), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL148), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL157), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL167), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL203-R208), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L42), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L56), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L137), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L148), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L157), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L167), [link](https://github.com/ant-design/ant-design/pull/42209/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739L203-R208))

### CODE REVIEW

File: components/timeline/style/index.ts [hunk 1]
```diff
@@ -3,15 +3,18 @@ import type { FullToken, GenerateStyle } from '../../theme/internal';
 import { genComponentStyleHook, mergeToken } from '../../theme/internal';
 import { resetComponent } from '../../style';
 
-export interface ComponentToken {}
+export interface ComponentToken {
+  color: string;
```
Comment (MadCcc) [inline components/timeline/style/index.ts:7]:
```suggestion
  tailColor: string;
```

File: components/timeline/style/index.ts [hunk 2]
```diff
@@ -3,15 +3,18 @@ import type { FullToken, GenerateStyle } from '../../theme/internal';
...
+  headBorderWidth: number;
+  headBg: string;
```
Comment (MadCcc) [inline components/timeline/style/index.ts:9]:
```suggestion
  dotBg: string;
```

File: docs/react/migrate-less-variables.en-US.md [hunk 3]
```diff
@@ -200,7 +194,17 @@ This document contains the correspondence between all the less variables related
 
 <!-- ### Tag -->
 
-<!-- ### Timeline -->
+### Timeline
+
+<!-- prettier-ignore -->
+| Less variables | Component Token | Note |
+| --- | --- | --- |
+| `@timeline-width` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:202]: 这个应该是 `tailWidth`，可以加回去

File: docs/react/migrate-less-variables.en-US.md [hunk 4]
```diff
@@ -200,7 +194,17 @@ This document contains the correspondence between all the less variables related
...
+| `@timeline-color` | `color` | - |
+| `@timeline-dot-border-width` | `headBorderWidth` | - |
+| `@timeline-dot-color` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:205]:
```suggestion
| `@timeline-dot-color` | - | Deprecated |
```
看起来 v4 都没有在用了，直接废弃

**Comment by MadCcc** [comment]
和上游冲突了

**Comment by MadCcc** [comment]
全局 format 了？需要把无关改动 revert 一下。

**Comment by jrr997** [comment]
合并提交的时候格式化了太多文件，不好revert，这里要重开了

**Pull request closed without merging**